### PR TITLE
Add forceRefill completion

### PR DESCRIPTION
**Pull request by alexfilimon**
Add completion on end reload data.

```swift
adapter?.forceRefill {
  // some code after reload ends
}
```

### CODE REVIEW

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 1]
```diff
@@ -323,6 +333,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+            return
+        }
+        let section = sourceIndexPath.section
+        var oldCellGenerators = cellGenerators[section]
+        let itemToMove = oldCellGenerators[sourceIndexPath.row]
+        oldCellGenerators.remove(at: sourceIndexPath.row)
+        oldCellGenerators.insert(itemToMove, at: destinationIndexPath.row)
+
+        let sectionGenerator = sectionHeaderGenerators[section]
+        removeAllGenerators(in: sectionGenerator)
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.removeAllGenerators(in: sectionGenerator)
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 2]
```diff
@@ -323,6 +333,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+        addCellGenerators(oldCellGenerators, toHeader: sectionGenerator)
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.addCellGenerators(oldCellGenerators, toHeader: sectionGenerator)
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 3]
```diff
@@ -323,6 +333,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+
+        cellChangedPosition.invoke(with: (section: section, oldIndex: sourceIndexPath.row, newIndex: destinationIndexPath.row))
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.cellChangedPosition.invoke(with: (section: section, 
                                                                         oldIndex: sourceIndexPath.row, 
                                                                         newIndex: destinationIndexPath.row))
```

File: Source/BaseTableDataDisplayManager/PaginableBaseTableDataDisplayManager.swift [hunk 4]
```diff
@@ -14,7 +14,10 @@ open class PaginableBaseTableDataDisplayManager: BaseTableDataDisplayManager {
     public var lastCellShowingEvent = BaseEvent<Void>()
 
     public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
-        if indexPath.row == cellGenerators[indexPath.section].count - 1 {
+        let lastSectionIndex = cellGenerators.count - 1
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/PaginableBaseTableDataDisplayManager.swift]:
```suggestion
        let lastSectionIndex = self.cellGenerators.count - 1
```

File: Source/BaseTableDataDisplayManager/PaginableBaseTableDataDisplayManager.swift [hunk 5]
```diff
@@ -14,7 +14,10 @@ open class PaginableBaseTableDataDisplayManager: BaseTableDataDisplayManager {
...
+        let lastCellInLastSectionIndex = cellGenerators[lastSectionIndex].count - 1
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/PaginableBaseTableDataDisplayManager.swift]:
```suggestion
        let lastCellInLastSectionIndex = self.cellGenerators[lastSectionIndex].count - 1
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 6]
```diff
@@ -111,6 +112,15 @@ extension BaseTableDataDisplayManager {
         self.tableView?.reloadData()
     }
 
+    public func forceRefill(completion: @escaping (() -> Void)) {
+        CATransaction.begin()
+        CATransaction.setCompletionBlock {
+            completion()
+        }
+        forceRefill()
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.forceRefill()
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift (hunk 3 above)
Comment (alexfilimon) [reply Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]: Now there are only two params (was 3), and I think the break to the next line isn't needed

**Pull request merged**